.github/docker/Dockerfile.musl

```diff
@@ -2,13 +2,19 @@
 ARG PLATFORM=arm64
 ARG NODE_VERSION=16.20.1
 
-FROM ${PLATFORM}/node:${NODE_VERSION}-alpine AS node
+FROM ${PLATFORM}/node:${NODE_VERSION}-alpine AS build
```

Comment:
What's the motivation behind this change?

Reply:
The last line in the dockerfile:
```
COPY --from=build /zstd/prebuilds/ /
```

Requires that tag used in --from is the same as the tag here.  I chose to change this to build to match Dockerfile.glibc instead of adjusting the tag below to `--from=node`.